src/mlpack/methods/ann/layer/linear3d.hpp

```diff
@@ -0,0 +1,179 @@
+/**
+ * @file methods/ann/layer/linear3d.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Definition of the Linear layer class which accepts 3D input.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_LINEAR3D_HPP
+#define MLPACK_METHODS_ANN_LAYER_LINEAR3D_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/regularizer/no_regularizer.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the Linear3D layer class. The Linear class represents a
+ * single layer of a neural network.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat,
+    typename RegularizerType = NoRegularizer
+>
+class Linear3D
+{
+ public:
+  //! Create the Linear3D object.
+  Linear3D();
+
+  /**
+   * Create the Linear3D layer object using the specified number of units.
+   *
+   * @param inSize The number of input units.
+   * @param outSize The number of output units.
+   * @param regularizer The regularizer to use, optional.
+   */
+  Linear3D(const size_t inSize,
+         const size_t outSize,
+         RegularizerType regularizer = RegularizerType());
```

Comment:
You need 2 extra spaces here:
```suggestion
  Linear3D(const size_t inSize,
           const size_t outSize,
           RegularizerType regularizer = RegularizerType());
```